knora/dsplib/utils/xml_upload.py

```diff
@@ -910,94 +883,435 @@ def xml_upload(input_file: str, server: str, user: str, password: str, imgdir: s
         if resource.ark:
             resource_iri = convert_ark_v0_to_resource_iri(resource.ark)
 
+        # in case of a multimedia resource: upload the multimedia file
         resource_bitstream = None
         if resource.bitstream:
-            img = sipi_server.upload_bitstream(os.path.join(imgdir, resource.bitstream.value))
+            img: dict[Any, Any] = try_network_action(
```

Comment:
the typing here is far from ideal (as it is both unspecific and not correct, it could also be None), but that can be improved at a later date.